lib/puppet/node/environment.rb

```diff
@@ -361,13 +358,11 @@ def modules_by_path
     modules_by_path = {}
     modulepath.each do |path|
       if Puppet::FileSystem.exist?(path)
-        Dir.chdir(path) do
-          module_names = Dir.entries(path).select do |name|
-            Puppet::Module.is_module_directory?(name, path)
-          end
-          modules_by_path[path] = module_names.sort.map do |name|
-            Puppet::Module.new(name, File.join(path, name), self)
-          end
+        module_names = Dir.entries(path).select do |name|
+          Puppet::Module.is_module_directory?(name, path)
+        end
+        modules_by_path[path] = module_names.sort.map do |name|
+          Puppet::Module.new(name, File.join(path, name), self)
```

Comment:
I don't believe the `Puppet::Module#initialize` method relies on the current working directory. We pass in `File.join(path, name)` as the path to the module. It's set as an instance variable on the module:

https://github.com/puppetlabs/puppet/blob/aa1df95ab3c8ee5bd8f007ba45cca422b1de2498/lib/puppet/module.rb#L73

And then is used to resolve the path to `metadata.json`:

https://github.com/puppetlabs/puppet/blob/aa1df95ab3c8ee5bd8f007ba45cca422b1de2498/lib/puppet/module.rb#L285-L290